internal/locate/region_request.go

```diff
@@ -1288,21 +1282,24 @@ func (s *replicaSelector) onServerIsBusy(
 	return true, nil
 }
 
-// For some reasons, the leader is unreachable by now, try followers instead.
-// the state is changed in accessFollower.next when leader is unavailable.
-func (s *replicaSelector) canFallback2Follower() bool {
+// canSkipServerIsBusyBackoff returns true if the request can be sent to next replica and can skip ServerIsBusy backoff.
+func (s *replicaSelector) canSkipServerIsBusyBackoff() bool {
 	if s == nil || s.state == nil {
 		return false
 	}
-	state, ok := s.state.(*accessFollower)
-	if !ok {
+	accessLeader, ok := s.state.(*accessKnownLeader)
+	if ok && accessLeader.isCandidate(s.replicas[accessLeader.leaderIdx]) {
+		// If leader is still candidate, the request will be sent to leader again,
+		// so don't skip since the leader is still busy.
 		return false
 	}
-	if !state.isStaleRead {
-		return false
+	for _, replica := range s.replicas {
+		if replica.attempts == 0 && replica.store.getLivenessState() == reachable &&
+			!replica.isEpochStale() && !replica.store.isSlow() {
+			return true
+		}
```

Comment:
Is it sure that the next attempt must be on such a node, no matter what selectorState it is currently and whether the state will change?

Reply:
Strictly speaking, this is not guaranteed. I can't call `state.next()` to get next replica directly, because `state.next()` method is stateful, call it will affect internal `lastIdx`.

But fortunately, it misjudgment at most once in extreme cases.

Following is a misjudgment example with `ReplicaReadMixed`:

1. Try replica-1, got server is busy, skip backoff since replica-3 is good, but replica-2's store is slow.
2. Try replica-2, got server is busy, skip backoff since replica-3 is good.
3. Try replica-3, return success.